### PR TITLE
[ja] translate console-and-shells.rst

### PR DESCRIPTION
**Pull request by okinaka**

### CODE REVIEW

File: ja/console-and-shells.rst [hunk 1]
```diff
@@ -1,18 +1,1094 @@
...
+    # server シェルの実行
+    bin/cake server
+
+    # migrations シェルの実行
+    bin/cake migrations -h
+
+    # bake の実行 (プラグインのプレフィックス付き)
+    bin/cake bake.bake -h
+
+シェルの名前が、アプリケーションやフレームワークのシェルとオーバーラップしていない場合、
```
Comment (chinpei215) [inline ja/console-and-shells.rst]: > オーバーラップ

`重複` would be better.

File: ja/console-and-shells.rst [hunk 2]
```diff
@@ -1,18 +1,1094 @@
...
+            $this->out(print_r($user, true));
+        }
+    }
+
+上記のシェルは、username によってユーザを取得し、データベースに格納された情報が表示されます。
+
+シェルのタスク
+==============
+
+より高機能なコンソールアプリケーションを開発するにあたり、ある程度時間的余裕がある場合は、
```
Comment (chinpei215) [inline ja/console-and-shells.rst]: > There will be times when building more advanced console applications,

In this sentence, `times` would mean `occasions`, not `enough time`.

より高度なコンソールアプリケーションを開発する場合には、

File: ja/console-and-shells.rst [hunk 3]
```diff
@@ -1,18 +1,1094 @@
...
+これらの機能を再利用可能なクラスとして再構成することで、これらを多くのシェル間で共有できます。
```
Comment (chinpei215) [inline ja/console-and-shells.rst]: >  you'll want to

多くのシェル間で共有される再利用可能なクラスとして機能を構成したいでしょう。

**Comment by okinaka** [comment]
@chinpei215 Thank you! I updated this PR.

File: ja/console-and-shells.rst [hunk 4]
```diff
@@ -1,18 +1,1094 @@
...
+       'extra' => [
+            'foo' => 'bar'
+        ]
+    ]);
+
+``extra`` で渡されたパラメータは、 ``Shell::$params`` プロパティにマージされ、
+``Shell::param()`` メソッドでアクセス可能になります。
+シェルで ``dispatchShell()`` を使用してディスパッチされた時、デフォルトで ``requested``
+追加パラメータが自動的に追加されます。この ``requested`` パラメータは、
+ディスパッチされたシェルに表示されている CakePHP のコンソールウェルカムメッセ​​ージを防ぎます。
```
Comment (chinpei215) [inline ja/console-and-shells.rst]: > ウェルカムメッセージ

Found two zero-width-spaces (Unicode: 200B / UTF-8: E2808B) between `メッセ` and `ージ`.
Probably, you can confirm it with VIM or some binary editor.

File: ja/console-and-shells.rst [hunk 5]
```diff
@@ -1,18 +1,1094 @@
...
+
+``$this->stdout->styles()`` を使ってさらに多くのスタイルを追加できます。
+新しいスタイルを追加するには以下のようにします。 ::
+
+    $this->_io->styles('flashy', ['text' => 'magenta', 'blink' => true]);
+
+これで ``<flashy>`` というタグが有効になり、ansi カラーが有効な端末であれば、
+``$this->out('<flashy>うわ！</flashy> 何か変になった');`` の場合の表示は
+色がマゼンタでブリンクになります。
+スタイルを定義する際は `text` と `background` 属性として以下の色が指定できます。
```
Comment (chinpei215) [inline ja/console-and-shells.rst]: In the English version, `text` and `background` are enclosed with double backquotes.

**Comment by chinpei215** [comment]
Great work! I'll merge this PR after test passing.

**Comment by chinpei215** [comment]
I'll close and re-open this PR to re-run test.

**Comment by chinpei215** [comment]
@okinaka The test has passed :)

**Pull request merged**